x-pack/solutions/observability/plugins/infra/server/lib/alerting/inventory_metric_threshold/lib/create_request.ts

```diff
@@ -21,13 +21,13 @@ import { createBucketSelector } from './create_bucket_selector';
 import { KUBERNETES_POD_UID, NUMBER_OF_DOCUMENTS, termsAggField } from '../../common/utils';
 
 const ADDITIONAL_CONTEXT_ALLOW_LIST = ['host.*', 'labels.*', 'tags', 'cloud.*', 'orchestrator.*'];
-const ADDITIONAL_CONTEXT_BLOCKED_LIST = ['host.cpu.*', 'host.disk.*', 'host.network.*'];
+export const ADDITIONAL_CONTEXT_BLOCKED_LIST = ['host.cpu.*', 'host.disk.*', 'host.network.*'];
 
 export const ADDITIONAL_CONTEXT_BLOCKED_LIST_REGEX = new RegExp(
   '^' +
-    ADDITIONAL_CONTEXT_BLOCKED_LIST.map((p) => p.replace(/\./g, '\\.').replace(/\*/g, '.*')).join(
-      '|'
-    ) +
+    ADDITIONAL_CONTEXT_BLOCKED_LIST.map((p) =>
```

Comment:
This is.. cryptic. And it was cryptic before, so this is not a critique of this PR or anyone that wrote it. Regex in general is not fun for most people to deal with. I think we should re-write this to be more intentional, and perhaps we should escape _all_ regex operators.

```suggestion
/**
 * This function escapes Regex special characters with a (\) so they will behave like normal text.
 * @param str input text
 * @returns input text with regex operators escaped
 * @example
 *   escapeRegex("foo.bar") -> "foo\.bar"
 *   escapeRegex("a+b?")    -> "a\+b\?"
 */
function escapeRegex(str: string) {
  return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
}

function wildcardToRegex(str: string) {
  return escapeRegex(str).replace(/\\\*/g, '.*');
}

export const ADDITIONAL_CONTEXT_BLOCKED_LIST_REGEX = new RegExp(
  `^(${ADDITIONAL_CONTEXT_BLOCKED_LIST.map(wildcardToRegex).join('|')})$`
);
```

This doesn't break any of the unit tests and I presume will be an ok revision. Even if we decide not to escape all operators, we should still move these to separate named functions to better express intention for the sake of people who will read this code in the future.